velero-plugin-for-aws/object_store_test.go

```diff
@@ -152,3 +164,85 @@ func TestValidChecksumAlg(t *testing.T) {
 		})
 	}
 }
+
+func TestObjectStore_PutObject(t *testing.T) {
+	type args struct {
+		bucket string
+		key    string
+		body   io.Reader
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+		wantLog string
+	}{
+		{
+			name:    "PutObject produces logs",
+			wantLog: "Authorization: AWS4-HMAC-SHA256",
+			args: args{
+				// bucket: "my-bucket",
+				bucket: "mybucket",
+				key:    "mykey",
+				body:   bytes.NewReader([]byte("my-data")),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			logger := logrus.New()
+			logger.Level = logrus.DebugLevel
+			var buffer bytes.Buffer
+			logger.Out = &buffer
+			awsLogger := logging.LoggerFunc(func(classification logging.Classification, format string, v ...interface{}) {
+				log := logger.WithFields(logrus.Fields{
+					"sdk":            "aws-sdk-go-v2",
+					"classification": classification,
+				})
+				log.Debugf(format, v...)
+			})
+
+			// create a fake s3 server
+			backend := s3mem.New()
+			faker := gofakes3.New(backend)
+			ts := httptest.NewServer(faker.Server())
+			defer ts.Close()
+			backend.CreateBucket(tt.args.bucket)
+
+			cfg, err := config.LoadDefaultConfig(
+				context.TODO(),
+				config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider("KEY", "SECRET", "SESSION")),
+				config.WithHTTPClient(&http.Client{
+					Transport: &http.Transport{
+						TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+					},
+				}),
+				config.WithEndpointResolverWithOptions(
+					aws.EndpointResolverWithOptionsFunc(func(_, _ string, _ ...interface{}) (aws.Endpoint, error) {
+						return aws.Endpoint{URL: ts.URL}, nil
+					}),
+				),
+				config.WithLogger(awsLogger),
+				config.WithClientLogMode(aws.LogRequest|aws.LogResponse|aws.LogRetries|aws.LogSigning),
```

Comment:
Please feedback on which [logs to enable](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/aws#ClientLogMode) and/or if we need to make this configurable.